csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs

```diff
@@ -14,7 +14,7 @@ public class SurveyResponse
 
         // <SnippetSurveyStatus>
         public bool AnsweredSurvey => surveyResponses != null;
-        public string Answer(int index) => surveyResponses.GetValueOrDefault(index, "No answer")!;
+        public string Answer(int index) => surveyResponses?.GetValueOrDefault(index) ?? "No answer";
```

Comment:
I think this should be

```suggestion
        public string Answer(int index) => surveyResponses?[index] ?? "No answer";
```

Reply:
@Thraka, index op on a dictionary even with a null conditional operator may result in a runtime `KeyNotFoundException`. Obviously, one could check for `ContainsKey` but the resulting code would be pretty verbose:
```suggestion
        public string Answer(int index) => surveyResponses?.GetValueOrDefault(index) ?? "No answer";
```
I simply left the extension method as it makes the code readable.

Reply:
This is a good discussion.  @paneer is correct, the extension method is needed.

I'll mark this conversation resolved.

Reply:
Ahh yes duh on me!